src/main/java/org/folio/rest/impl/EholdingsPackagesImpl.java

```diff
@@ -186,22 +189,55 @@ public void getEholdingsPackages(String filterCustom, String q, String filterSel
   public void postEholdingsPackages(String contentType, PackagePostRequest entity, Map<String, String> okapiHeaders,
                                     Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     packagesPostBodyValidator.validate(entity);
+    RMAPITemplate template = templateFactory.createTemplate(okapiHeaders, asyncResultHandler);
 
     PackagePost packagePost = packagePostRequestConverter.convert(entity);
+    String accessTypeId = entity.getData().getAttributes().getAccessTypeId();
 
-    templateFactory.createTemplate(okapiHeaders, asyncResultHandler)
-      .requestAction(context ->
+    if (accessTypeId == null) {
+      template.requestAction(context ->
         getVendorId(context)
           .thenCompose(id -> context.getPackagesService().postPackage(packagePost, id))
-          .thenApply(packageById -> new PackageResult(packageById, null, null))
-      )
+          .thenApply(packageById -> new PackageResult(packageById, null, null)));
+    } else {
+      template.requestAction(context ->
+        accessTypesService.existsById(accessTypeId, okapiHeaders)
+          .thenCompose(aBoolean -> {
+            if (Boolean.TRUE.equals(aBoolean)) {
+              return getVendorId(context)
+                .thenCompose(id -> context.getPackagesService().postPackage(packagePost, id))
+                .thenApply(packageById -> new PackageResult(packageById, null, null))
+                .thenCompose(packageResult -> updateAccessType(accessTypeId, packageResult, okapiHeaders));
+            } else {
+              throw new ValidationException("Access type does not exist");
```

Comment:
Could you use InputValidationException instead of ValidationException, please?

Reply:
and use constant for an error message like ACCESS_TYPE_NOT_FOUND_MESSAGE as it probably will be used in the future PUT method

Reply:
Fixed.